docs/includes/fundamentals/aggregation/AggregationsBuilderTest.php

```diff
@@ -84,27 +92,235 @@ public function testAggregationBuilderProjectStage(): void
         $this->assertArrayNotHasKey('_id', $result->first());
     }
 
-    public function testAggregationBuilderPipeline(): void
+    public function testAggregationBuilderMatchGroup(): void
     {
-        // begin pipeline example
-        $pipeline = User::aggregate()
-            ->addFields(
-                birth_year: Expression::year(
-                    Expression::dateFieldPath('birthday'),
-                ),
+        Sale::truncate();
+
+        Sale::insert([
+            [
+                '_id' => 1,
+                'item' => 'abc',
+                'price' => new Decimal128('10'),
+                'quantity' => 2,
+                'date' => new UTCDateTime(new DateTimeImmutable('2014-03-01T08:00:00Z')),
+            ],
+            [
+                '_id' => 2,
+                'item' => 'jkl',
+                'price' => new Decimal128('20'),
+                'quantity' => 1,
+                'date' => new UTCDateTime(new DateTimeImmutable('2014-03-01T09:00:00Z')),
+            ],
+            [
+                '_id' => 3,
+                'item' => 'xyz',
+                'price' => new Decimal128('5'),
+                'quantity' => 10,
+                'date' => new UTCDateTime(new DateTimeImmutable('2014-03-15T09:00:00Z')),
+            ],
+            [
+                '_id' => 4,
+                'item' => 'xyz',
+                'price' => new Decimal128('5'),
+                'quantity' => 20,
+                'date' => new UTCDateTime(new DateTimeImmutable('2014-04-04T11:21:39.736Z')),
+            ],
+            [
+                '_id' => 5,
+                'item' => 'abc',
+                'price' => new Decimal128('10'),
+                'quantity' => 10,
+                'date' => new UTCDateTime(new DateTimeImmutable('2014-04-04T21:23:13.331Z')),
+            ],
+            [
+                '_id' => 6,
+                'item' => 'def',
+                'price' => new Decimal128('7.5'),
+                'quantity' => 5,
+                'date' => new UTCDateTime(new DateTimeImmutable('2015-06-04T05:08:13Z')),
+            ],
+            [
+                '_id' => 7,
+                'item' => 'def',
+                'price' => new Decimal128('7.5'),
+                'quantity' => 10,
+                'date' => new UTCDateTime(new DateTimeImmutable('2015-09-10T08:43:00Z')),
+            ],
+            [
+                '_id' => 8,
+                'item' => 'abc',
+                'price' => new Decimal128('10'),
+                'quantity' => 5,
+                'date' => new UTCDateTime(new DateTimeImmutable('2016-02-06T20:20:13Z')),
+            ],
+        ]);
+
+        // start-builder-match-group
+        $pipeline = Sale::aggregate()
+            ->match(
+                date: [
+                    Query::gte(new UTCDateTime(new DateTimeImmutable('2014-01-01'))),
+                    Query::lt(new UTCDateTime(new DateTimeImmutable('2015-01-01'))),
```

Comment:
I wish we could skip the `new UTCDateTime` part, but that's not possible due to type restrictions. https://jira.mongodb.org/browse/PHPLIB-1634